src/utilities/typedQueryDocumentNode.d.ts

```diff
@@ -9,6 +9,10 @@ export interface TypedQueryDocumentNode<
 > extends DocumentNode {
   readonly definitions: ReadonlyArray<ExecutableDefinitionNode>;
   // FIXME: remove once TS implements proper way to enforce nominal typing
-  readonly __enforceStructuralTypingOnResponseDataType?: TResponseData;
-  readonly __enforceStructuralTypingOnRequestVariablesType?: TRequestVariables;
+  /**
+   * This type is used to ensure that the variables you pass in to the query are assignable to Variables
+   * and that the Result is assignable to whatever you pass your result to. The method is never actually
+   * implemented, but the type is valid because we list it as optional
+   */
+  __workaroundForTSToEnsureVariablesAndResultMatching?: (variables: TRequestVariables) => TResponseData;
```

Comment:
@ardatan I think adding the word "workaround" into the name is too much, how about something like this? (feel free to improve spelling or use better wording)
```suggestion
  __ensureTypesOfVariablesAndResultMatching?: (variables: TRequestVariables) => TResponseData;
```